jenkins_pipelines/environments/common/pipeline-reference.groovy

```diff
@@ -0,0 +1,44 @@
+def run(params) {
+    timestamps {
+        deployed = false
+        env.resultdir = "${WORKSPACE}/results"
+        env.resultdirbuild = "${resultdir}/${BUILD_NUMBER}"
+        env.common_params = "--outputdir ${resultdir} --tf ${params.tf_file} --gitfolder ${resultdir}/sumaform"
+        try {
+            stage('Clone sumaform and susemanager-ci') {
+                // Create a directory for all the outputs belonging to this build number
+                sh "mkdir -p ${resultdir}"
+                git url: params.terracumber_gitrepo, branch: params.terracumber_ref
+                dir("susemanager-ci") {
+                    checkout scm
+                }
+                // Create a directory for  to place the directory with the build results (if it does not exist)
+                sh "mkdir -p ${resultdir}"
+                // Clone sumaform
+                sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${common_params} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync"
+            }
+            stage('Deploy') {
+                // Provision the environment
+                if (params.terraform_bin_init) {
+                    env.TERRAFORM_INIT = '--init'
+                } else {
+                    env.TERRAFORM_INIT = ''
+                }
+                sh "set +x; source /home/jenkins/.credentials set -x; TERRAFORM=${params.terraform_bin} TERRAFORM_PLUGINS=${params.terraform_bin_plugins} ./terracumber-cli ${common_params} --logfile ${resultdirbuild}/sumaform.log ${env.TERRAFORM_INIT} --taint '.*(domain|main_disk).*' --runstep provision"
+                deployed = true
+            }
+        }
+        finally {
+            stage('Get results') {
+                if (!deployed) {
+                  // Send email
+                  sh "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/mail.log --runstep mail"
+                }
+                // Clean up old results
+                sh "./clean-results -r ${resultdir}"
```

Comment:
I'm not sure how this script behaves, it will do the clean up of the workspace?
We only want a clean up of the workspace if the pipeline had a fully success, as otherwise we might need some details in the workspace.
As we do here https://github.com/SUSE/susemanager-ci/blob/master/jenkins_pipelines/manager_testsuite/Jenkinsfile_qa_testsuite#L110

Reply:
No, this cleans the old results (so we only have the last 20 results). I think I am going to change the script name.

As far as I could see, no Jenkins job had that parameter `cleanWorkspace`, so that part never run.

But even so, that's dangerous. The tfstate is on the workspace, we should not clean it.

Reply:
Yep, you might be right, maybe we are doing wrong currently.

Reply:
Script renamed. Done.